### ISSUE TITLE
LuceneIndexing does not honour if a parent ContentType is deleted or unpublished

### ISSUE THREAD

**Issue by micrub-it** [OrchardCMS/OrchardCore#8639]
I have a contentItem of a contentType with a ListPart. If I delete (or unpublish) a ‘parent’ contentItem, a lucene Index to ListPart type still list a list parts.
For example, I have two blogs and a lucene index for blog posts with "Latest Version Index" unselected. If I delete (or unpublish) a blog, the lucene index still returns the posts of the deleted (or unpublished) blog.

### Expected behavior
Lucene Indexing should not list a ListPart items of a deleteted or unpublished parent ContentItem

**Comment by Skrypt** [comment]
So basically when we unpublish a content item that has a list part it should unpublish also all the content items added to this list part. What if I'm using that same content item on 2 different list parts?

**Comment by deanmarcussen** [comment]
I think this was implemented in O1 recently - unpublished of contained items when the list is unpublished.

I don’t think you can share them between lists @Skrypt 

So yes, we think this feature could be implemented

**Comment by Skrypt** [comment]
Right now you can't because you can't assign a content item to a list without creating it in that list.
But, in 01 we could pick a content item to add it to a list. There is no constraint either in the code that says here this content item can't be in those 2 lists.

**Comment by Skrypt** [comment]
So basically what I do is a custom handler to remove these. That's what they are meant for and this is per use case. But if we would want to add this to the ListPart then it could be optional per Content Type that contains that ListPart. This is not something that I would like to work everytime. I could do lists of content items just to group them together and reorder them. These could be different lists ordered differently per culture for example.

**Comment by deanmarcussen** [comment]
The `ListPart` / `ContainedPart` would constrain it I think @Skrypt - while we don't have move functionality currently, if we did, the `ContainedPart` would blow up if you tried to link two items to different parents. 

Agree that it might be a setting.

**Comment by Skrypt** [comment]
![image](https://user-images.githubusercontent.com/3228637/108762811-8d98c480-751e-11eb-9b63-ce53ad662eea.png)

**Comment by Skrypt** [comment]
@deanmarcussen This ListContentItemId should have been an array : 

```json
  "ContainedPart": {
    "ListContentItemId": "4dftc2g2s4t6mtn4e16w0ajg4y",
    "Order": 0
  },
```

**Comment by micrub-it** [comment]
> I think this was implemented in O1 recently - unpublished of contained items when the list is unpublished.
> 
> I don’t think you can share them between lists @Skrypt
> 
> So yes, I think this feature could be implemented



> unpublish

I think the normal use of list part is to define single-parent children. However, in any case, the unpublish / deletion of the child content item may simply not take place if the child has another parent not unpublished / deleted

**Comment by Skrypt** [comment]
So, what you want is that when a Blog is unpublished that his BlogPost be also unpublished. But what if you had already specific items that we're drafts in this Blog "container". One could want to "restore" the Blog with the content items in the status they we're before unpublishing it. It needs some thinking because, I'm using currently a custom handler to unpublish some related Content Items and I do this for only certain content types. Maybe what we need is something more generic to make a relation between different content types. Here, the ListPart is just a use case. But right now in OC you generally define these relations by extending the current IContentHandler events.

**Comment by deanmarcussen** [comment]
> I think the normal use of list part is to define single-parent children. However, in any case, the unpublish / deletion of the child content item may simply not take place if the child has another parent not unpublished / deleted

I agree. I have no trouble with a pr adding a handler for this. It could be based on a setting (optional).

@Skrypt I think your use case has moved into custom handler, rather than the specific issue @micrub-it is discussing

**Comment by Skrypt** [comment]
I don't mind a handler for ListPart but honestly I think it should be optional.

**Comment by deanmarcussen** [comment]
Setting is good

> only if we can't add existing content items to a List or have content items in multiple lists.

This would always be another part not a change to List, that would be too breaking.

(but really taxonomies is the feature for this)
